imap_processing/idex/idex_cdf_attrs.py

```diff
@@ -1,173 +1,177 @@
+    "contains the data from a single dust impact. See "
+    "https://imap.princeton.edu/instruments/idex for more details."
+)
+
+idex_base = InstrumentBase(__version__, "IDEX>Interstellar Dust Experiment", text)
+
+idex_l1_global_attrs = DataLevelBase(
+    "L1>Level-1",
+    "imap_idex_l1",
+    "IMAP Mission IDEX Instrument Level-1 Data.",
+    idex_base,
+)
+
+idex_l2_global_attrs = DataLevelBase(
+    "L2>Level-2", "imap_idex_l2", "IMAP Mission IDEX Instrument Level-2 Data", idex_base
+)
+
+l1_data_base = ScienceBase(
+    IdexConstants.DATA_MIN,
+    IdexConstants.DATA_MAX,
+    display_type="spectrogram",
+    depend_0="Epoch",
+    format="I12",
+    units="dN",
+    var_type="data",
+    variable_purpose="PRIMARY",
+)
 
 # L1 variables base dictionaries
 # (these need to be filled in by the variable dictionaries below)
-l1_data_base = {
-    "DEPEND_0": "Epoch",
-    "DISPLAY_TYPE": "spectrogram",
-    "FILLVAL": cdf_utils.INT_FILLVAL,
-    "FORMAT": "I12",
-    "UNITS": "dN",
-    "VALIDMIN": DATA_MIN,
-    "VALIDMAX": DATA_MAX,
-    "VAR_TYPE": "data",
-    "SCALETYP": "linear",
-    # "VARIABLE_PURPOSE" tells CDAWeb which variables are worth plotting
-    "VARIABLE_PURPOSE": "PRIMARY",
-}
-
-l1_tof_base = {"DEPEND_1": "Time_High_SR"} | l1_data_base
-
-l1_target_base = {"DEPEND_1": "Time_Low_SR"} | l1_data_base
-
-sample_rate_base = {
-    "DEPEND_0": "Epoch",
-    "FILLVAL": cdf_utils.DOUBLE_FILLVAL,
-    "FORMAT": "F12.5",
-    "LABLAXIS": "Time",
-    "UNITS": "microseconds",
-    "VALIDMIN": SAMPLE_RATE_MIN,
-    "VALIDMAX": SAMPLE_RATE_MAX,
-    "VAR_TYPE": "support_data",
-    "SCALETYP": "linear",
-    "VAR_NOTES": (
+l1_tof_base = dataclasses.replace(l1_data_base, depend_1="Time_High_SR")
+l1_target_base = dataclasses.replace(l1_data_base, depend_1="Time_Low_SR")
+
+sample_rate_base = FloatBase(
+    IdexConstants.SAMPLE_RATE_MIN,
+    IdexConstants.SAMPLE_RATE_MAX,
+    depend_0="Epoch",
+    format="F12.5",
+    label_axis="Time",
+    units="microseconds",
+    var_notes=(
         "The number of microseconds since the event.  "
         "0 is the start of data collection, negative "
         "numbers represent data collected prior to a dust event"
     ),
-}
-
-trigger_base = {
-    "DEPEND_0": "Epoch",
-    "FILLVAL": cdf_utils.INT_FILLVAL,
-    "FORMAT": "I12",
-    "VALIDMIN": 0,  # All values are positive integers or 0 by design
-    "VAR_TYPE": "data",
-    "DISPLAY_TYPE": "no_plot",
-}
+)
+
+# TODO: is "VALIDMAX" not required here? Added INT_MAXVAL, incl the existing CDF for ref
+trigger_base = IntBase(
+    0,
+    GlobalConstants.INT_MAXVAL,
+    depend_0="Epoch",
+    format="I12",
+    var_type="data",
+    display_type="no_plot",
+)
+#     {
+#     "DEPEND_0": "Epoch",
+#     "FILLVAL": Constants.INT_FILLVAL,
+#     "FORMAT": "I12",
+#     "VALIDMIN": 0,  # All values are positive integers or 0 by design
+#     "VAR_TYPE": "data",
+#     "DISPLAY_TYPE": "no_plot",
+# }
```

Comment:
Do you need this for later?

Reply:
I wasn't sure of the answer, it should be answered probably by the IDEX developer since it's related to the specific IDEX attribute

Reply:
It was commented out. That's why I was wondering. cool